demos/dotnet/dotnet/NugetAction.cs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+
+namespace dotnet
+{
+    internal static class NugetAction
+    {
+        public static bool GetNugetAndRestore(ProjectProperties properties, Log log)
+        {
```

Comment:
This method seems like it's really this:

    {
        EnsureNugetExe();
        RestoreProjectDependencies(properties);
    }

Is that the right way to factor it?